### PR TITLE
Add option to disable trailing commas on multi-line collections

### PR DESCRIPTION
**Pull request by mlavergn**
Extends the configuration options to allow disabling the addtion of trailing commas on multi-line collections.

### CODE REVIEW

File: Sources/SwiftFormat/API/Configuration.swift [hunk 1]
```diff
@@ -162,6 +163,9 @@ public struct Configuration: Codable, Equatable {
   /// Contains exceptions for the `NoAssignmentInExpressions` rule.
   public var noAssignmentInExpressions: NoAssignmentInExpressionsConfiguration
 
+  /// Determines whether multi-line list initializers should have trailing commas
```
Comment (allevato) [inline Sources/SwiftFormat/API/Configuration.swift:166]: Can you end this sentence with a period?

File: Tests/SwiftFormatTests/PrettyPrint/CommaTests.swift [hunk 2]
```diff
@@ -0,0 +1,145 @@
+import SwiftFormat
+
+final class CommaTests: PrettyPrintTestCase {
```
Comment (allevato) [inline Tests/SwiftFormatTests/PrettyPrint/CommaTests.swift:3]: Can you add some tests that cover dictionary literals as well?

Comment (mlavergn) [reply Tests/SwiftFormatTests/PrettyPrint/CommaTests.swift:3]: Note that tests for single-line dictionary setters wrap differently than single-line array setters, that's reflected in the expected test output.

**Comment by mlavergn** [comment]
I gleaned the name `multilineCollectionTrailingCommas` from an older PR, but since this does affect single line dictionaries, I wonder if `collectionTrailingCommas` would be more appropriate.

**Comment by allevato** [comment]
> I gleaned the name `multilineCollectionTrailingCommas` from an older PR, but since this does affect single line dictionaries, I wonder if `collectionTrailingCommas` would be more appropriate.

Yeah, "line" is the wrong word here, because the logic isn't based on whether the collection wraps across multiple lines, it's about whether it has multiple elements—single element arrays/dictionaries never have a trailing comma inserted, even if they wrap (because of Swift parser subtleties around how collection types are parsed in expression contexts). I think there are other places in swift-format where we use "line" when we really mean "statement" or "element", but fixing those up require ensuring compatibility with older configs.

How do you feel about `multiElementCollectionTrailingCommas`?

**Comment by mlavergn** [comment]
> > I gleaned the name `multilineCollectionTrailingCommas` from an older PR, but since this does affect single line dictionaries, I wonder if `collectionTrailingCommas` would be more appropriate.
> 
> Yeah, "line" is the wrong word here, because the logic isn't based on whether the collection wraps across multiple lines, it's about whether it has multiple elements—single element arrays/dictionaries never have a trailing comma inserted, even if they wrap (because of Swift parser subtleties around how collection types are parsed in expression contexts). I think there are other places in swift-format where we use "line" when we really mean "statement" or "element", but fixing those up require ensuring compatibility with older configs.
> 
> How do you feel about `multiElementCollectionTrailingCommas`?

It's a mouthful, but it certainly self-documents what the setting is for. I'll refactor it.

**Comment by mlavergn** [comment]
Refactored the setting name to `multiElementCollectionTrailingCommas` and use `literal` rather than `initializer` in the docs.

**Comment by mlavergn** [comment]
Added a DocC header to auto-gen proper documentation based on the styling of `indentSwitchCaseLabels`

**Review by allevato — APPROVED** [review]
This looks great. Thanks for your contribution!

**Pull request merged**